### DISCUSSION TITLE
Leverage Core Accounting Principles when Modeling Financial Data

### DISCUSSION THREAD

**Discussion by fivetran-joemarkiewicz** [dbt-labs/docs.getdbt.com#1311] [Archive]
# Overview
Posting here to continue the thread from the [dbt-technical-blog-writing](https://github.com/dbt-labs/dbt-technical-blog-writing/discussions/44) discussion on In Depth Accounting Modeling. I was presented with the opportunity to start a discussion in this repo to propose a possible blog post on the topic. As I have had the joy of working with a lot of dbt packages modeling accounting data in a standardized format, I thought it would be great to take a "retro" and informative approach to this blog. 

I wanted to share my thoughts on an outline for this possible blog here and to get your initial thoughts before writing anything more. I have included the outline below, but also will share my [Google Doc](https://docs.google.com/document/d/1S9xE_IJeE49IyhYu2CsgYgXFzUDpUTzbpTWEg00jNKw/edit) with the outline included so you may add more detailed comments if you would like. Let me know what think, and what next steps for writing a first draft of the blog post would be!
> - If you don't have access to the Google Doc, feel free to request access and I will approve as soon as possible.
> - The name is super lengthy and very much open for that to change to something better!

# Outline
**Intro:**
- Accounting and data modeling are not so different
   - Highlight Joel’s tweet on this subject!
   - Highlight the core accounting principles and how they are consistently applied for effective financial analysis
   - Regardless of what accounting software you are using, these standards will be consistently applied to how you report on your financial data
- Highlight why it is beneficial for analysts to leverage these standardized models when doing their financial analysis
   - Core principle of packages, “why rebuild the wheel when someone else has done the work for you already” 
   - You can leverage these principles regardless of what data loader you are using and what accounting software you are leveraging
- Touch on the past dbt packages developed (Netsuite, QuickBooks, Xero, Sage Intacct) that apply these accounting principles and the three types of output models they provide:
   - General Ledger / Transaction Detail
   - Balance Sheet
   - Profit & Loss Statement

**Body:**
- Transaction focused general ledger as the base of all reporting
   - How to build the general ledger or transaction detail equivalent
   - Share what are some common key learnings and mistakes when building the GL
- General Ledger provided by period for better monthly breakdowns and rolling sums for balance sheet accounts
   - Go over the SQL that has been used in Sage, QuickBooks, and Xero to explode the GL for monthly reporting
- Easy filtering from the period GL for financial statements
   - Profit and Loss
   - Balance Sheet
- How to effectively leverage these end models in the wild
   - What more can you analyze with these output models

**Conclusion:**
- Quick recap on what was learned when building out the accounting packages and how the standardization was successful
- Conclude on how you can leverage the pre-built models from the packages or apply the same principles to your own modeling
- Ponder thoughts for the future on what other sources can leverage standardized modeling principles and why this is exciting
   - Customer Support (Zendesk, Intercom, Jira)
   - Billing and Subscriptions (Stripe, ReCharge, Recurly)
   - Marketing (Marketo, Mailchimp, Iterable, Kalviyo)

**Comment by dpmp-canary** [comment]
Hey @fivetran-joemarkiewicz - did you end up writing this post? (: Think the outline is great
